includes/REST_API/Products_Controller.php

```diff
@@ -0,0 +1,489 @@
+	 */
+	protected function get_items_shopify( $request ) {
+		/**
+		 * Search term.
+		 *
+		 * @var string $search_term
+		 */
+		$search_term = $request['search'];
+
+		/**
+		 * Filters the shopify products data TTL value.
+		 *
+		 * @since 1.20.0
+		 *
+		 * @param int $time Time to live (in seconds). Default is 1 day.
+		 * @param WP_REST_Request $request Full details about the request.
+		 */
+		$cache_ttl = apply_filters( 'web_stories_shopify_data_cache_ttl', DAY_IN_SECONDS, $request );
+		$cache_key = ! empty( $request['search'] ) ? 'web_stories_shopify_data_' . md5( $search_term ) : 'web_stories_shopify_data_default';
+
+		$data = get_transient( $cache_key );
+
+		if ( \is_string( $data ) && ! empty( $data ) ) {
+			/**
+			 * Decoded cached products data.
+			 *
+			 * @var array|null $products
+			 */
+			$products = json_decode( $data, true );
+
+			if ( $products ) {
+				return rest_ensure_response( $products );
+			}
+		}
+
+		$results = [];
+
+		/**
+		 * Shopify host.
+		 *
+		 * @var string $host
+		 */
+		$host = $this->settings->get_setting( Settings::SETTING_NAME_SHOPIFY_HOST );
+
+		/**
+		 * Shopify access token.
+		 *
+		 * @var string $access_token
+		 */
+		$access_token = $this->settings->get_setting( Settings::SETTING_NAME_SHOPIFY_ACCESS_TOKEN );
+
+		// TODO(#11154): Maybe move to a class constant.
+		$api_version = '2022-01';
+
+		// TODO(#11154): Proper sanitization.
+		$url = sprintf(
+			'https://%1$s/api/%2$s/graphql.json',
+			$host,
+			$api_version
+		);
+
+		$search_string = empty( $search_term ) ? '*' : '*' . $search_term . '*';
+
+		// TODO(#11154): Support different sortKeys.
+		$query = <<<QUERY
+{
+  products(first: 100, sortKey: CREATED_AT, query: "available_for_sale:true AND title:$search_string") {
+    edges {
+      node {
+        id
+        title
+        vendor
+        description
+        priceRange {
+          minVariantPrice {
+            amount
+            currencyCode
+          }
+        }
+        onlineStoreUrl
+        images(first: 10) {
+          edges {
+            node {
+              url
+              altText
+            }
+          }
+        }
+      }
+    }
+  }
+}
+QUERY;
+
+		$response = wp_remote_post(
+			$url,
+			[
+				'headers' => [
+					'Content-Type'                      => 'application/graphql',
+					'X-Shopify-Storefront-Access-Token' => $access_token,
+				],
+				'body'    => $query,
+			]
+		);
+
+		if ( WP_Http::OK !== wp_remote_retrieve_response_code( $response ) ) {
+			return new \WP_Error( 'rest_unknown', __( 'Error fetching products', 'web-stories' ), [ 'status' => 404 ] );
+		}
+
+		/**
+		 * Shopify GraphQL API response.
+		 *
+		 * @var ShopifyGraphQLResponse $result
+		 */
+		$result = json_decode( wp_remote_retrieve_body( $response ), true );
+
+		// TODO: Error handling.
+		if ( isset( $result['errors'] ) ) {
+			return new \WP_Error( 'rest_unknown', __( 'Error fetching products', 'web-stories' ), [ 'status' => 404 ] );
+		}
```

Comment:
We should not swallow all errors here. Can you at least see what the error and return different errors in our API.

Reply:
Yes, that's why there is a TODO for this here.

See also https://github.com/GoogleForCreators/web-stories-wp/pull/11158#discussion_r842238474